### ISSUE TITLE
termsrv.dll 10.0.19041.1949 [Not supported]

### ISSUE THREAD

**Issue by adream001** [stascorp/rdpwrap#1968]
Could someone make it done? the dll file is attached.
[termsrv.zip](https://github.com/stascorp/rdpwrap/files/9437586/termsrv.zip)

**Comment by SirRufo** [comment]
I guess it will not get fixed here ... but [there](https://github.com/sebaxakerhtc/rdpwrap.ini/issues/144)

**Comment by Duck15** [comment]
为啥我会收到这条邮件捏？？？

发送自 Windows 10 版邮件应用

发件人: Sir Rufo
发送时间: 2022年8月27日 19:23
收件人: stascorp/rdpwrap
抄送: Subscribed
主题: Re: [stascorp/rdpwrap] termsrv.dll 10.0.19041.1949 [Not supported](Issue #1968)

I guess it will not get fixed here ... but there

**Comment by affinityv** [comment]
[termsrv--Win10--21H1--2022-08-28--10.0.19041.1949--64bit.zip](https://github.com/stascorp/rdpwrap/files/9438200/termsrv--Win10--21H1--2022-08-28--10.0.19041.1949--64bit.zip)

**Comment by affinityv** [comment]
Refer https://github.com/sebaxakerhtc/rdpwrap.ini/issues/144

**Comment by affinityv** [comment]
2022-08 Cumulative Update Preview for Windows 10 Version 21H1 for x64-based Systems (KB5016688)

It should be fixed for here... so, here is a 7z file saved as a zip file which has both the 64bit termsrv.dll and the pdb files for @sebaxakerhtc to perform his magic

[termsrv--Win10--21H1--2022-08-28--10.0.19041.1949--64bit.zip](https://github.com/stascorp/rdpwrap/files/9438208/termsrv--Win10--21H1--2022-08-28--10.0.19041.1949--64bit.zip)

**Comment by affinityv** [comment]
Updates done by our very good friend! @sebaxakerhtc 
[rdpwrap.ini](https://raw.githubusercontent.com/affinityv/INI-RDPWRAP/master/rdpwrap.ini)

**Comment by gamma-577** [comment]
> Updates done by our very good friend! @sebaxakerhtc [rdpwrap.ini](https://raw.githubusercontent.com/affinityv/INI-RDPWRAP/master/rdpwrap.ini%5D)

Hi, 

Sorry, can't find the rdpwrap.ini file. Do you have offsets? I feel they are a better and cleaner solution than replacing the termsrv.dll. Thanks. :-)

-Gamma

**Comment by gamma-577** [comment]
> Updates done by our very good friend! @sebaxakerhtc [rdpwrap.ini](https://raw.githubusercontent.com/affinityv/INI-RDPWRAP/master/rdpwrap.ini%5D)

Hi, 

Got the offsets here : https://github.com/sebaxakerhtc/rdpwrap.ini/commit/a5e8c6d7990d0a98e2ba13e2a644ce5b828f5355

Thanks!

-Gamma

**Comment by affinityv** [comment]
> > Updates done by our very good friend! @sebaxakerhtc [rdpwrap.ini](https://raw.githubusercontent.com/affinityv/INI-RDPWRAP/master/rdpwrap.ini%5D)
> 
> Hi,
> 
> Sorry, can't find the rdpwrap.ini file. Do you have offsets? I feel they are a better and cleaner solution than replacing the termsrv.dll. Thanks. :-)
> 
> -Gamma

Okay, there was a trailing square bracket with the link, that's all -- the rdpwrap.ini file is there; I've fixed my comment above.

**Comment by LorenzoTecnab** [comment]
Not working on Windows 10 32 bit even if the rdpinfo tell me fully supported, but only 1 rdp connection is allowed (i receive the warning of user already connected that will be disconnected)... what is the problem?

**Comment by affinityv** [comment]
> Not working on Windows 10 32 bit even if the rdpinfo tell me fully supported, but only 1 rdp connection is allowed (i receive the warning of user already connected that will be disconnected)... what is the problem?

Please upload the 32 bit termsrv.dll file so that, hopefully, the offsets can be found from it.

**Comment by WillianPeterson** [comment]
Sorry about my bad brazillian´s english, but i cant access local printers ...

**Comment by affinityv** [comment]
> 

Looks like you posted in the wrong thread.

**Comment by affinityv** [comment]
> Sorry about my bad brazillian´s english, but i cant access local printers ...

Looks like you posted in the wrong thread.

**Comment by LorenzoTecnab** [comment]
> > Not working on Windows 10 32 bit even if the rdpinfo tell me fully supported, but only 1 rdp connection is allowed (i receive the warning of user already connected that will be disconnected)... what is the problem?
> 
> Please upload the 32 bit termsrv.dll file so that, hopefully, the offsets can be found from it.

Here my termsrv.dll (32 bit W10)

[termsrv.zip](https://github.com/stascorp/rdpwrap/files/9450344/termsrv.zip)

**Comment by sebaxakerhtc** [comment]
> Here my termsrv.dll (32 bit W10)

The offsets was added - feel free to try it out and let we know if it work or not for you
https://github.com/sebaxakerhtc/rdpwrap.ini/commit/cbd7dfeeb56eecea8e86928313ca6608d346a1fc

**Comment by LorenzoTecnab** [comment]
> > Here my termsrv.dll (32 bit W10)
> 
> The offsets was added - feel free to try it out and let we know if it work or not for you [sebaxakerhtc/rdpwrap.ini@cbd7dfe](https://github.com/sebaxakerhtc/rdpwrap.ini/commit/cbd7dfeeb56eecea8e86928313ca6608d346a1fc)

Thanks! Now it's working correctly!!

**Comment by WillianPeterson** [comment]
> > Sorry about my bad brazillian´s english, but i cant access local printers ...
> 
> Looks like you posted in the wrong thread.

I agree with you...it was weird, sorry about that. But my haste in sending the message and copying and pasting from Google Translate made this wrong impression.
I wanted to say that although I have inserted the new rdpwrapper.ini, suggested on this page, I am accessing the remote printers normally, but I cannot print to the local printers.

**Comment by affinityv** [comment]
> I wanted to say that although I have inserted the new rdpwrapper.ini, suggested on this page, I am accessing the remote printers normally, but I cannot print to the local printers.

Could you access the local printers form the remote previously?  I think, edit the .rdp file and save it.  Perhaps if you don't have a special .rdp, then the one in your "Documents" directory ... ```%USERPROFILE%\Documents\Default.rdp```

**Comment by WillianPeterson** [comment]
> > I wanted to say that although I have inserted the new rdpwrapper.ini, suggested on this page, I am accessing the remote printers normally, but I cannot print to the local printers.
> 
> Could you access the local printers form the remote previously? I think, edit the .rdp file and save it. Perhaps if you don't have a special .rdp, then the one in your "Documents" directory ... `%USERPROFILE%\Documents\Default.rdp`

Yes, I was previously able to access all local printers from the remote computer.
Comparing files I found, in the last rpdwraper.ini, this session after the [PatchCodes] session:

[SLInit]
bServerSku=1
bRemoteConnAllowed=1
bFUSEnabled=1
bAppServerAllowed=1
bMultimonAllowed=1
lMaxUserSessions=0
ulMaxDebugSessions=0
bInitialized=1

After adding these lines of code everything worked normally again.

**Comment by affinityv** [comment]
> After adding these lines of code everything worked normally again.

Okay, those values are the same as my [rdpwrap.ini](https://raw.githubusercontent.com/affinityv/INI-RDPWRAP/master/rdpwrap.ini) file.

**Comment by laggingreflex** [comment]
[The fix](https://github.com/stascorp/rdpwrap/issues/1968#issuecomment-1229174316) also worked for my version: 19044.1949 (as opposed to 19041.1949 in the title), so leaving it for anyone searching.

**Comment by sebaxakerhtc** [comment]
> for my version: 19044.1949

You look at Windows version, but should look at termsrv.dll version

**Comment by zekecarter** [comment]
@sebaxakerhtc I have left 2 videos so you can see the bug and the integration of the solution, I hope you can include it in the new version of rdpwrap.ini

https://www.filemail.com/d/sesloezlxtsefzb

I have uploaded 2 videos to filemail, the first part shows the bug and where I got it from.

the second video shows how I integrated the solution and how it works

I hope it is of your help and thank you for providing excellent assistance greetings

**Comment by affinityv** [comment]
@zekecarter .... the "Windows version" is irrelevant; the termsrv.dll version is what is important.  RDPConf.exe shows the correct termsrv.dll version.  It is normal behaviour and procedure to do an install, then update the rdpwrap.ini file and then do a restart of the service.  What is it you are trying to show with the videos?

**Comment by zekecarter** [comment]
> @zekecarter .... the "Windows version" is irrelevant; the termsrv.dll version is what is important. RDPConf.exe shows the correct termsrv.dll version. It is normal behaviour and procedure to do an install, then update the rdpwrap.ini file and then do a restart of the service. What is it you are trying to show with the videos?

I was just trying to show that when I installed the rdpwrapper program it did not install the update for the termsrv.dll version I have and that the rdpwrap.ini when checking the 19041.1949 version I already had it but it did not work, and I had to copy another version I found from sebaxakerhtc to include it to the rdpwrap.ini I have and I thought they could integrate it to the new version of rdpwrap.ini.

thousand apologies

**Comment by narggari** [comment]
![Desktop Screenshot 2022 09 16 - 20 14 36 83](https://user-images.githubusercontent.com/113712058/190700552-e686a095-35e6-4d8a-9d6b-5259d82e3c1a.png)
why does it say this?

**Comment by zekecarter** [comment]
> ![Desktop Screenshot 2022 09 16 - 20 14 36 83](https://user-images.githubusercontent.com/113712058/190700552-e686a095-35e6-4d8a-9d6b-5259d82e3c1a.png) why does it say this?

Hello, this same error happened to me. I have windows 10 x64 and this solution worked for me:

https://github.com/sebaxakerhtc/rdpwrap.ini/commit/a5e8c6d7990d0a98e2ba13e2a644ce5b828f5355

**Comment by GTAdoum** [comment]
Hello,
I have version 19041.1949 and the patch from zekecarter, or even the one found dated 09/17a, do not work, even if the status shows fully supported.

**Comment by yesidtaz** [comment]
https://github.com/Octanium91/rdpwupDownloads/blob/master/rdpwup_8.zip

**Comment by crasher32** [comment]
> > Here my termsrv.dll (32 bit W10)
> 
> The offsets was added - feel free to try it out and let we know if it work or not for you [sebaxakerhtc/rdpwrap.ini@cbd7dfe](https://github.com/sebaxakerhtc/rdpwrap.ini/commit/cbd7dfeeb56eecea8e86928313ca6608d346a1fc)

Thanks! Now it's working correctly!!
